pkg/httpclient/client.go

```diff
@@ -24,13 +26,18 @@ type Option func(opts *Options)
 
 // Options are configuration options for an HTTP client.
 type Options struct {
-	Timeout       time.Duration
-	TLS           *tls.Config
-	Logger        *logrus.Logger
-	ACSToken      string
-	CheckRedirect func(req *http.Request, via []*http.Request) error
+	Timeout         time.Duration
+	TLS             *tls.Config
+	Logger          *logrus.Logger
+	ACSToken        string
+	CheckRedirect   func(req *http.Request, via []*http.Request) error
+	FailOnErrStatus bool
 }
 
+type ctxKey int
```

Comment:
`ctxKey` and `ctxKeyFailOnErrStatus` are private but I have trouble seeing their purpose without commands, can you add some?

Reply:
I've added some comments. Also as a side note the custom type is the recommended way to go in such cases, from https://golang.org/pkg/context/#WithValue : 
> Users of WithValue should define their own types for keys.